File: class04/cpp/AP0403-Exemplo02.cpp
#include <iostream>
#include <string>
int main() {
    std::string sa, sb;
    std::cout << "Circuito XOR usando portas básicas:" << std::endl;
    std::cout << "XOR = (A OR B) AND NOT (A AND B)" << std::endl;
    std::cout << "A (true/false): "; std::cin >> sa;
    std::cout << "B (true/false): "; std::cin >> sb;
    bool a = (sa == "true"), b = (sb == "true");
    bool xr = (a || b) && !(a && b);
    std::cout << "A XOR B = " << std::boolalpha << xr << std::endl;
    return 0;
}

File: class04/java/AP0403-Exemplo02.java
import java.util.Scanner;
public class DesafioLogicoPortas {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Circuito XOR usando portas básicas:");
        System.out.println("XOR = (A OR B) AND NOT (A AND B)");
        System.out.print("A (true/false): ");
        boolean a = sc.nextBoolean();
        System.out.print("B (true/false): ");
        boolean b = sc.nextBoolean();
        boolean xor = (a || b) && !(a && b);
        System.out.println("A XOR B = " + xor);
        sc.close();
    }
}